extensions/master_extension/v1/win-e2e-master-extension.sh

```diff
@@ -1,5 +1,13 @@
 #!/bin/bash
 
+# Wait for all pods in kube-system namespace to be running before applying taints
+
+OUT=$(kubectl get pods -n kube-system -o custom-columns=STATUS:status.phase | grep -v STATUS | grep -v Running)
+while [[ ! -z $OUT ]]; do 
+sleep 2
+OUT=$(kubectl get pods -n kube-system -o custom-columns=STATUS:status.phase | grep -v STATUS | grep -v Running)
+done

```

Comment:
nits:

this also works, and it's a bit cleaner, IMO:

    kubectl get pods -n kube-system -o custom-columns=STATUS:status.phase | grep -v "STATUS\|Running"

additionally, the code inside the while block could be indented.

also, question: if there's something wrong with the deployment itself and some pods will always be in the Pending state, will this script be timed out somehow?

Reply:
aks-engine times it out, however it may be too much of a wait and we can time it out here.

Reply:
@adelina-t  when you make the changes asked by @bclau  i can approve